src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -248,8 +271,27 @@ private void RaiseNoLivingNodesAlert(string alertMsg, string dbName)
             foreach (var member in topology.Members)
             {
                 var status = None;
-                if (current.TryGetValue(member, out var nodeStats) &&
-                    nodeStats.Status == ClusterNodeStatusReport.ReportStatus.Ok &&
+                if(current.TryGetValue(member, out var nodeStats) == false)
+                {
+                    // there isn't much we can do here, except for log it.
+                    if (previous.TryGetValue(member, out _))
+                    {
+                        // if we found this node in the previous report, we will ingore it this time and wait for the next report.
+                        continue;
+                    }
+
+                    var msg =
+                        $"The member node {member} was not found in both current and previuos reports of the cluster observer. " +
+                        $"If this error continue to raise, check the latancy between the cluster nodes.";
+                    if (_logger.IsInfoEnabled)
```

Comment:
What about if there is an error? And how can we avoid spamming the log about this?

Reply:
Since we generate an empty report for a node that haven't sent any info yet.
So this actually should be a very rare case when there is a race between adding the nodes to `ClusterMaintenanceSupervisor` and analyzing the stats in the observer.